File: accidentes/AMarotoChivite-Accidentes/AMarotoChivite-Accidentes-Java/src/main/java/Ficheros/Accidentes/utils/ReadDataOfFile.java
package Ficheros.Accidentes.utils;

import Ficheros.Accidentes.config.ConfigApp;
import Ficheros.Accidentes.models.dto.AccidenteDto;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import static java.util.stream.Collectors.toList;

public class ReadDataOfFile {
    public List<AccidenteDto> readDataOfCSV() {
        String localFile = ConfigApp.APP_DATA + File.separator + "accidentes.csv";
        File file = new File(localFile);

        List<String> lines = new ArrayList<>();

        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
            String line;
            // Eliminamos la primera línea
            int lineNumber = 0;
            while ((line = br.readLine()) != null) {
                if (lineNumber != 0) {
                    lines.add(line);
                }
                lineNumber++;
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        List<AccidenteDto> listAccidenteDto = lines.stream().map(line -> {
            String[] fields = line.split(";");

            String numExpediente = fields[0];
            String fecha = fields[1];
            String hora = fields[2];
            String localizacion = fields[3];
            String numero = fields[4];

            // Filtrar las filas que no tienen un valor para el campo "cod_distrito".
            // El índice 5 corresponde al campo "cod_distrito".
            String codDistrito = "";
            if (fields[5] == "") {
                codDistrito = "nulo";
            } else {
                codDistrito = fields[5];
            }

            String distrito = fields[6];
            String tipoAccidente = fields[7];
            String estadoMeteorologico = fields[8];
            String tipoVehiculo = fields[9];
            String tipoPersona = fields[10];
            String rangoEdad = fields[11];
            String sexo = fields[12];
            String codLesividad = fields[13];
            String lesividad = fields[14];
            String coordenadaX = fields[15];
            String coordenadaY = fields[16];
            String positivoAlcohol = fields[17];
            String positivoDroga = fields[18];

            return new AccidenteDto(numExpediente, fecha, hora, localizacion, numero, codDistrito, distrito, tipoAccidente, estadoMeteorologico, tipoVehiculo, tipoPersona, rangoEdad, sexo, codLesividad, lesividad, coordenadaX, coordenadaY, positivoAlcohol, positivoDroga);
        }).collect(toList());

        return listAccidenteDto;
    }
}

File: accidentes/AMarotoChivite-Accidentes/AMarotoChivite-Accidentes-Kotlin/src/main/kotlin/Ficheros/Accidentes/utils/readDataOfFile.kt
package Ficheros.Accidentes.utils

import Ficheros.Accidentes.config.ConfigApp
import Ficheros.Accidentes.models.dto.AccidenteDTO
import java.io.File

fun readDataOfCSV(): List<AccidenteDTO> {
    val localFile = "${ConfigApp.APP_DATA}${File.separator}accidentes.csv"
    val file = File(localFile)

    // Leer el fichero completo y eliminamos la primera fila
    val lines = file.readLines().drop(1)

    val listAccidenteDTO = lines.map { line ->

        val fields = line.split(';')
        val numExpediente = fields[0]
        val fecha = fields[1]
        val hora = fields[2]
        val localizacion = fields[3]
        val numero = fields[4]

        // Filtrar las filas que no tienen un valor para el campo "cod_distrito".
        // El índice 5 corresponde al campo "cod_distrito".
        var codDistrito = ""
        if (fields[5] == "") {
            codDistrito = "nulo"
        } else {
            codDistrito = fields[5]
        }

        val distrito = fields[6]
        val tipoAccidente = fields[7]
        val estadoMeteorologico = fields[8]
        val tipoVehiculo = fields[9]
        val tipoPersona = fields[10]
        val rangoEdad = fields[11]
        val sexo = fields[12]
        val codLesividad = fields[13]
        val lesividad = fields[14]
        val coordenadaX = fields[15]
        val coordenadaY = fields[16]
        val positivoAlcohol = fields[17]
        val positivoDroga = fields[18]


        AccidenteDTO(
            numExpediente,
            fecha,
            hora,
            localizacion,
            numero,
            codDistrito,
            distrito,
            tipoAccidente,
            estadoMeteorologico,
            tipoVehiculo,
            tipoPersona,
            rangoEdad,
            sexo,
            codLesividad,
            lesividad,
            coordenadaX,
            coordenadaY,
            positivoAlcohol,
            positivoDroga
        )
    }
    return listAccidenteDTO
}